src/newbank/server/NewBank.java

```diff
@@ -22,15 +22,30 @@ public synchronized Customer checkLogInDetails(String customerID, String passwor
 	}
 
 	// commands from the NewBank customer are processed in this method
-	public synchronized String processRequest(String customerID, String request) {
-		Customer customer = customers.getCustomer(customerID);
-		if( customer != null ) {
+	public synchronized String processRequest(String userID, String request) {
+		User user = users.getUser(userID);
+		if( user != null ) {
+			
+			//Execute SHOWMYACCOUNTS if the user is a customer
 			if (request.equals("SHOWMYACCOUNTS")) {
-				return showMyAccounts(customer );
+				if (user.getUserType().equals("customer")) {
+					Customer customer = (Customer) user;
+					return showMyAccounts( customer );					
+				}
+			//Execute NEWACCOUNT if the user is a customer			
 			} else if (request.startsWith("NEWACCOUNT")) {
-				String accountName = request.substring( request.indexOf(" ")+1);
-				return makeNewAccount(customer, accountName);
+				if (user.getUserType().equals("customer")) {
+					Customer customer = (Customer) user;
+					String accountName = request.substring( request.indexOf(" ")+1);
+					return makeNewAccount(customer, accountName);				
+				}
+			//Execute SHOWACCOUNTS if the user is a customer			
+			} else if (request.startsWith("SHOWACCOUNTS")) {
+				if (user.getUserType().equals("banker")) {
```

Comment:
Yeah true.  I can see a lot more like this which can be put in backlog specific to Banker 
Depositing money also could be something which can be Banker specific